src/app/ReadClient.cpp

```diff
@@ -541,9 +541,18 @@ CHIP_ERROR ReadClient::ProcessEventReportIBs(TLV::TLVReader & aEventReportIBsRea
 CHIP_ERROR ReadClient::RefreshLivenessCheckTimer()
 {
     CancelLivenessCheckTimer();
+    System::Clock::Timeout timeout =
+        std::chrono::duration_cast<std::chrono::milliseconds>(System::Clock::Seconds16(mMaxIntervalCeilingSeconds));
```

Comment:
```suggestion
    System::Clock::Timeout timeout = System::Clock::Seconds16(mMaxIntervalCeilingSeconds);
```
You need to cast explicitly if losing precision, but this is not losing precision.  The types involved know how to convert the thing on the right (known to be in seconds) to the thing on the left (known to be in milliseconds).